src/common/centile.cc

```diff
@@ -0,0 +1,106 @@
+/*
+ * centile.cc
+ *
+ *  Created on: 25-Aug-2016
+ *      Author: prashant.kr
+ */
+#include "centile.h"
+
+namespace centile {
+  Centile::Centile(unsigned int start, unsigned int end, unsigned int inc) {
+    this->start = start;
+    this->end = end;
+    this->inc = inc;
+
+    numBuckets = (end-start)/inc + 1;
+    buckets = new atomic64_t[numBuckets];
+    firstBucket = start/inc;
+    lastBucket = end/inc;
+  }
+
+  Centile::~Centile() {
+    delete buckets;
+  }
+
+  atomic64_t& Centile::get_sample_count() {
+    return sample_count;
+  }
+
+  void Centile::insert(unsigned int value) {
+    unsigned int index = int(value/inc);
```

Comment:
what happens when value is negative? you'll be casting a signed int to an unsigned int. It might accidentally fall into a valid bucket